### PR TITLE
Javadoc the enchantments package

### PR DESCRIPTION
**Pull request by NoComment1105**
This commit adds some javadocs to the enchantment package. Please let me know if i used any british spellings anywhere :) and feel free to yell if i make any giant mistakes

Also updates Gradle but htat can be removed if need be

### CODE REVIEW

File: mappings/net/minecraft/enchantment/EnchantmentTarget.mapping [hunk 1]
```diff
@@ -1,3 +1,4 @@
 CLASS net/minecraft/unmapped/C_lkwvnptq net/minecraft/enchantment/EnchantmentTarget
+	COMMENT This class specifes all the fields for each part type of tool/armour in the game
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/EnchantmentTarget.mapping]:
```suggestion
	COMMENT This class specifes all the fields for each part type of tool/armor in the game.
```

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 2]
```diff
@@ -178,6 +182,7 @@ CLASS net/minecraft/unmapped/C_jakrppis net/minecraft/enchantment/EnchantmentHel
 		ARG 1 entity
 			COMMENT the entity to choose equipments from
 	METHOD m_ysxfxsjj onTargetDamaged (Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_astfners;)V
+		COMMENT This method is called when the user of the Item with that enchantment damages another Entity ({@code target})}}
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]:
```suggestion
		COMMENT This method is called when the user of the Item with the enchantment damages another Entity ({@code target}).
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 3]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 		ARG 3 level
 	METHOD m_tiaspbpv getEquipment (Lnet/minecraft/unmapped/C_usxaxydn;)Ljava/util/Map;
 		ARG 1 entity
 	METHOD m_vfhytrxr getOrCreateTranslationKey ()Ljava/lang/String;
+		COMMENT {@return the translation key or create one if there isn't one}
 	METHOD m_vssdpxdz getTranslationKey ()Ljava/lang/String;
+		COMMENT {@return gets key for translating this in the lang files}
 	METHOD m_zfqdifee getName (I)Lnet/minecraft/unmapped/C_rdaqiwdt;
+		COMMENT {@return a translation key that is formatted to a specific color}
+		COMMENT By defualt it changes enchantments that {@return true} in {@code isCursed()} to have red names.
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT By default it changes enchantments that {@return true} in {@code isCursed()} to have red names.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 4]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
+		COMMENT {@return the minimum level required to by the player for the enchantment to be shown in the enchantment table.}
+		COMMENT It is not a 1:1 ratio. The majority of enchantments return something similar to `10 * level`, often with different scales depending on the maximum level and rarity of the enchantment.
 		ARG 1 level
 	METHOD m_smaihzzt isAvailableForEnchantedBookOffer ()Z
 		COMMENT {@return whether this enchantment will appear in the enchanted book trade
 		COMMENT offers of librarian villagers}
 	METHOD m_sufqbdfo isCursed ()Z
+		COMMENT {@return weather the enchantment is a curse. E.G: Curse of Binding}
 	METHOD m_sxyjtpkf onUserDamaged (Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_astfners;I)V
+		COMMENT This method is called when the Player ({@code user}), who is wearing something that contains the enchantment, is attacked by another Entity ({@code attacker})}
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT This method is called when the Player ({@code user} is attacked by another Entity ({@code attacker}) and is wearing something that contains the enchantment}.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 5]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 		ARG 3 level
 	METHOD m_rtwmhoyh getMinPower (I)I
+		COMMENT {@return the minimum level required to by the player for the enchantment to be shown in the enchantment table.}
+		COMMENT It is not a 1:1 ratio. The majority of enchantments return something similar to `10 * level`, often with different scales depending on the maximum level and rarity of the enchantment.
 		ARG 1 level
 	METHOD m_smaihzzt isAvailableForEnchantedBookOffer ()Z
 		COMMENT {@return whether this enchantment will appear in the enchanted book trade
 		COMMENT offers of librarian villagers}
 	METHOD m_sufqbdfo isCursed ()Z
+		COMMENT {@return weather the enchantment is a curse. E.G: Curse of Binding}
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT {@return whether the enchantment is a curse. E.G: Curse of Binding}
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 6]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
+		COMMENT {@return the type of item ({@code stack}) that can have this applied to it}
 		ARG 1 stack
 	METHOD m_kzbinilx getAttackDamage (ILnet/minecraft/unmapped/C_amnulxyk;)F
+		COMMENT {@return the amount of extra damage to apply to the entity group specified with {@code group}.}
 		ARG 1 level
 		ARG 2 group
 	METHOD m_lkqpuckl isTreasure ()Z
+		COMMENT {@return wheather this enchantment will appear in the enchantment table and loot chests or loot chests only. E.G Mending}
 	METHOD m_phdcatdd onTargetDamaged (Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_astfners;I)V
+		COMMENT This method is called when an Entity ({@code target}) is hit by the Player ({@code user}) with an Item that has this enchantment on it
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT This method is called when an Entity ({@code target}) is hit by the Player ({@code user}) with an Item that has this enchantment on it.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 7]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 	METHOD m_zfqdifee getName (I)Lnet/minecraft/unmapped/C_rdaqiwdt;
+		COMMENT {@return a translation key that is formatted to a specific color}
+		COMMENT By defualt it changes enchantments that {@return true} in {@code isCursed()} to have red names.
 		ARG 1 level
 	METHOD m_ztiddpby isAvailableForRandomSelection ()Z
 		COMMENT {@return whether this enchantment will appear in the enchanting table or
 		COMMENT loots with random enchant function}
 	METHOD m_ztycenho getMaxPower (I)I
+		COMMENT {@return the maximum level a Player is allowed before the enchant is not shown in the enchantment table}
+		COMMENT The default return is the minimum power level + 5
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT The default return is the minimum power level + 5.
```

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 8]
```diff
@@ -28,6 +28,7 @@ CLASS net/minecraft/unmapped/C_jakrppis net/minecraft/enchantment/EnchantmentHel
 		ARG 0 existing
 		ARG 1 candidate
 	METHOD m_gfmalqgf onUserDamaged (Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_astfners;)V
+		COMMENT This method is called when the user of the Item with that enchantment is damaged
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]:
```suggestion
		COMMENT This method is called when the user of the Item with the enchantment is damaged.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 9]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
 		ARG 0 id
 	METHOD m_fecfwekl getRarity ()Lnet/minecraft/unmapped/C_jxtrubuh$C_dnkiztld;
 	METHOD m_glfcpyqa getMaxLevel ()I
+		COMMENT Sets the maximum number of levels the enchantment has. Default: 1
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT Sets the maximum number of levels the enchantment has. Default: 1.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 10]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 	METHOD m_igxrorrr getMinLevel ()I
+		COMMENT Sets the minimum number of levels the enchantment has. Default: 1
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT Sets the minimum number of levels the enchantment has. Default: 1.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 11]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 		ARG 1 other
 	METHOD m_khmhdqpw isAcceptableItem (Lnet/minecraft/unmapped/C_sddaxwyk;)Z
+		COMMENT {@return the type of item ({@code stack}) that can have this applied to it}
 		ARG 1 stack
 	METHOD m_kzbinilx getAttackDamage (ILnet/minecraft/unmapped/C_amnulxyk;)F
+		COMMENT {@return the amount of extra damage to apply to the entity group specified with {@code group}.}
 		ARG 1 level
 		ARG 2 group
 	METHOD m_lkqpuckl isTreasure ()Z
+		COMMENT {@return wheather this enchantment will appear in the enchantment table and loot chests or loot chests only. E.G Mending}
```
Comment (OroArmor) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]:
```suggestion
		COMMENT {@return whether this enchantment will appear in the enchantment table and loot chests or loot chests only. E.G Mending}
```

**Review by OroArmor — CHANGES_REQUESTED** [review]
Make sure all comments end with a `.`. This is not for `{@return}`s or other tags, just the raw comments. I think I got them all too.

File: mappings/net/minecraft/enchantment/EnchantmentTarget.mapping (hunk 1 above)
Comment (NoComment1105) [reply mappings/net/minecraft/enchantment/EnchantmentTarget.mapping]: funni british moment

**Comment by NoComment1105** [comment]
All done with those @OroArmor

**Review by OroArmor — APPROVED** [review]

**Comment by OroArmor** [comment]
Oh make sure that you are consistent on using `that` or `the` when referring to an enchantment

**Comment by NoComment1105** [comment]
o7

**Comment by NoComment1105** [comment]
I believe i have them all 👍🏼

**Review by Bubblie01 — APPROVED** [review]
Good job Comment everything is accurate and good

File: mappings/net/minecraft/enchantment/EnchantmentTarget.mapping [hunk 12]
```diff
@@ -1,3 +1,4 @@
 CLASS net/minecraft/unmapped/C_lkwvnptq net/minecraft/enchantment/EnchantmentTarget
+	COMMENT This class specifes all the fields for each part type of tool/armor in the game.
```
Comment (IotaBread) [inline mappings/net/minecraft/enchantment/EnchantmentTarget.mapping]:
```suggestion
	COMMENT This class specifies fields for each part type of tool/armor in the game.
```

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 13]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 		ARG 1 other
 	METHOD m_khmhdqpw isAcceptableItem (Lnet/minecraft/unmapped/C_sddaxwyk;)Z
+		COMMENT {@return the type of item ({@code stack}) that can have this applied to it}
 		ARG 1 stack
 	METHOD m_kzbinilx getAttackDamage (ILnet/minecraft/unmapped/C_amnulxyk;)F
+		COMMENT {@return the amount of extra damage to apply to the entity group specified with {@code group}.}
 		ARG 1 level
 		ARG 2 group
 	METHOD m_lkqpuckl isTreasure ()Z
+		COMMENT {@return whether this enchantment will appear in the enchantment table and loot chests or loot chests only. E.G Mending}
```
Comment (IotaBread) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:34]: You should add that villagers can also sell these enchantments, but they sell it at twice the price they would with a normal enchantment

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 14]
```diff
@@ -15,44 +15,61 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
...
 		ARG 3 level
 	METHOD m_rtwmhoyh getMinPower (I)I
+		COMMENT {@return the minimum level required to by the player for the enchantment to be shown in the enchantment table.}
+		COMMENT It is not a 1:1 ratio. The majority of enchantments return something similar to `10 * level`, often with different scales depending on the maximum level and rarity of the enchantment.
 		ARG 1 level
 	METHOD m_smaihzzt isAvailableForEnchantedBookOffer ()Z
 		COMMENT {@return whether this enchantment will appear in the enchanted book trade
 		COMMENT offers of librarian villagers}
 	METHOD m_sufqbdfo isCursed ()Z
+		COMMENT {@return whether the enchantment is a curse. E.G: Curse of Binding}
```
Comment (IotaBread) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:49]: You should add that curse enchantments can't be removed from an item, even with a grindstone

**Review by IotaBread — COMMENTED** [review]
Looks good, I have a few comments

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 13 above)
Comment (NoComment1105) [reply mappings/net/minecraft/enchantment/Enchantment.mapping:34]: Fixed

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 14 above)
Comment (NoComment1105) [reply mappings/net/minecraft/enchantment/Enchantment.mapping:49]: Done

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 15]
```diff
@@ -15,44 +15,63 @@ CLASS net/minecraft/unmapped/C_jxtrubuh net/minecraft/enchantment/Enchantment
 		ARG 0 id
 	METHOD m_fecfwekl getRarity ()Lnet/minecraft/unmapped/C_jxtrubuh$C_dnkiztld;
 	METHOD m_glfcpyqa getMaxLevel ()I
+		COMMENT Sets the maximum number of levels the enchantment has. Default: 1.
```
Comment (Jamalam360) [inline mappings/net/minecraft/enchantment/Enchantment.mapping]: What's meant by levels here? XP?

**Review by Jamalam360 — COMMENTED** [review]
Just curious about that comment I left, other than that looks all good

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 15 above)
Comment (OroArmor) [reply mappings/net/minecraft/enchantment/Enchantment.mapping]: @NoComment1105 clearing this up will make this pr ready to merge. I want to get as much in before 1.18 is released.

Comment (Jamalam360) [reply mappings/net/minecraft/enchantment/Enchantment.mapping]: Feel free to merge without, it was more of a curiosity than a concern

**Review by Jamalam360 — APPROVED** [review]

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 15 above)
Comment (NoComment1105) [reply mappings/net/minecraft/enchantment/Enchantment.mapping]: Yes Levels of XP sorry i missed this i'll change it now

Comment (NoComment1105) [reply mappings/net/minecraft/enchantment/Enchantment.mapping]: wait not levels of xp i'm dumb, tiers as in Sharpness 1 Sharpness 5

**Comment by NoComment1105** [comment]
@OroArmor @Jamalam360 Just updated that last one, should be ready for merge

**Pull request merged**